WalletWasabi.Fluent/Helpers/CoinPocketHelper.cs

```diff
@@ -71,5 +71,5 @@ public static IEnumerable<(SmartLabel SmartLabel, ICoinsView Coins)> GetPockets(
 		return pockets;
 	}
 
-	public static IEnumerable<Pocket> GetPockets(this Wallet wallet) => wallet.Coins.GetPockets(wallet.AnonScoreTarget).Select(x => new Pocket(x));
+	public static IEnumerable<Pocket> GetPockets(this Wallet wallet) => wallet.Coins.Available().GetPockets(wallet.AnonScoreTarget).Select(x => new Pocket(x));
```

Comment:
This should be optional, and the filtering logic should be applied where the actual logic is.

```c#
public static IEnumerable<Pocket> GetPockets(this Wallet wallet, bool onlyAvailableCoins = false) => 
		wallet.Coins.GetPockets(wallet.AnonScoreTarget, onlyAvailableCoins).Select(x => new Pocket(x));
```
And in the other `GetPockets()` method, in the beginning, you can add a filtering condition:
```c#
public static IEnumerable<(SmartLabel SmartLabel, ICoinsView Coins)> GetPockets(this ICoinsView allCoins, int privateAnonSetThreshold, bool onlyAvailableCoins = false)
{
	if (onlyAvailableCoins)
	{
		allCoins = allCoins.Available();
	}

	...
}
```

What do you think? After this, the usage of the GetPockets() would be easy.

Reply:
This is much better, we can keep the old behaviour, too. I will go with this.